airgun/entities/bookmark.py

```diff
@@ -0,0 +1,81 @@
+from airgun.entities.base import BaseEntity
+from airgun.navigation import NavigateStep, navigator
+from airgun.views.bookmark import BookmarkEditView, BookmarksView
+
+
+class BookmarkEntity(BaseEntity):
+
+    # Note: creation procedure takes place on specific entity page, generic
+    # helper is implemented inside :class:`BaseEntity`.
+
+    def delete(self, entity_name, controller=None):
+        """Delete existing bookmark"""
+        view = self.navigate_to(self, 'All')
+        row_query = {'name': entity_name}
```

Comment:
That looks more readable for me, but it is just my opinion
```
query = entity_name
if controller:
    query = 'name = "{}" and controller = "{}"'.format(
        entity_name, controller)
view.search(query)
row_query = {'name': entity_name}
if controller:
    row_query['controller'] = controller
view.table.row(**row_query)['Actions'].widget.click(handle_alert=True)
```

Reply:
more over :)
```python
row_query = {'name': entity_name}
query = 'name = "{}"'.format(entity_name)
if controller:
    query = '{} and controller = "{}"'.format(query, controller)
    row_query['controller'] = controller

view.search(query)
view.table.row(**row_query)['Actions'].widget.click(handle_alert=True)
```

Reply:
@oshtaier can't agree 2 exactly the same conditions are better than one.
@ldjebran i was thinking the same way, thought rewriting the same var isn't that optimal approach. But maybe more readable, will update, thanks for suggestion.

Reply:
That will be fine even OS dev :)